x/gov/router.go

```diff
@@ -0,0 +1,46 @@
+package gov
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/cosmos/cosmos-sdk/x/gov/proposal"
+)
+
+// Router is a map from string to proposal.Handler
+// copied and modified from baseapp/router.go
+type Router interface {
+	AddRoute(r string, h proposal.Handler) (rtr Router)
+	Route(path string) (h proposal.Handler)
+}
+
+type router struct {
+	routes map[string]proposal.Handler
+}
+
+var _ Router = (*router)(nil)
+
+// Constructs new router
+func NewRouter() *router {
+	return &router{
+		routes: make(map[string]proposal.Handler),
+	}
+}
+
+var isAlphaNumeric = regexp.MustCompile(`^[a-zA-Z0-9]+$`).MatchString
+
+func (rtr *router) AddRoute(path string, h proposal.Handler) Router {
+	if !isAlphaNumeric(path) {
+		panic("route expressions can only contain alphanumeric characters")
+	}
+	if rtr.routes[path] != nil {
+		panic(fmt.Sprintf("route %s has already been initialized", path))
+	}
+
+	rtr.routes[path] = h
+	return rtr
+}
+
+func (rtr *router) Route(path string) proposal.Handler {
+	return rtr.routes[path]
```

Comment:
should we add a `HasRoute` function to check if the route is present and include it inside this function?